src/coreclr/nativeaot/Runtime/eventpipe/ep-rt-aot.cpp

```diff
@@ -399,12 +399,10 @@ ep_rt_thread_id_t
 ep_rt_aot_current_thread_get_id (void)
 {
     STATIC_CONTRACT_NOTHROW;
-
-#ifdef TARGET_UNIX
-    return static_cast<ep_rt_thread_id_t>(PalGetCurrentOSThreadId());
-#else
-    return static_cast<ep_rt_thread_id_t>(::GetCurrentThreadId ());
-#endif
+    // Current thread is available at this stage since EventPipe is initiating this call.
+    ep_rt_thread_handle_t thread_handle = ThreadStore::GetCurrentThreadIfAvailable();
+    EP_ASSERT (thread_handle != NULL);
+    return thread_handle->GetPalThreadIdForLogging();
```

Comment:
The behavior of the original implementation matches the CoreCLR implementation. This implementation is diverging from CoreCLR (it returns `pthread_self` instead of `syscall(SYS_gettid)`/`pthread_threadid_np`). I do not think this is a good change.

Ideally, we would have one implementation of the event pipe thread ID that is shared between all runtimes for given OS.